cylc/flow/suite_files.py

```diff
@@ -559,6 +554,92 @@ def register(reg=None, source=None, redirect=False):
     return reg
 
 
+def clean(reg):
+    """Remove stopped workflows on the local scheduler filesystem.
+
+    Deletes the run dir in ~/cylc-run and any symlink dirs. Note: if the
+    run dir has been manually deleted, it will not be possible to clean the
+    symlink dirs.
+
+    Args:
+        reg (str): workflow name.
+    """
+    _validate_reg(reg)
+    reg = os.path.normpath(reg)
+    if reg.startswith('.'):
```

Comment:
BTW the suite name validator has you covered here:

https://github.com/cylc/cylc-flow/blob/80bb0503692c3c5292e3980b5f07bda4606aeae1/cylc/flow/unicode_rules.py#L190

Reply:
The difference is I'm checking after normalising the path, so you could have `cylc clean foo/../..` which corresponds to `rm $HOME`. However, I guess I could change the order to
```python
reg = os.path.normpath(reg)
_validate_reg(reg)
```

Reply:
I don't think we should accept path manipulations, though I'm unaware of a simple method to block them (testing for `..` doesn't work for all cases).

Probs good enough to ensure that the flow exists within the cylc-run dir:

```
reg = Path(reg)
reg.resolve()
try:
    reg.relative_to(run_dir_path)
except ValueError:
    raise Exception('Flow not found')
```

Anyway, not a problem for this PR.